### PR TITLE
Refactor hitpoint function

### PR DESCRIPTION
**Pull request by Njord201**
I refactored the hitPoint function, changed the return type from Math::Point3D to Color (new class) and renamed the function to getColotPoint

### CODE REVIEW

File: include/Color.hpp [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+*/
+
+#pragma once
+
+class Color {
+
+    public:
+
+        /**
+         * @brief Construct a new Color object
```
Comment (AxelF44) [inline include/Color.hpp:15]:
```suggestion
         * @brief Construct a new Color object.
```

File: include/Color.hpp [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+         *
+         * @param r red value
+         * @param g green value
+         * @param b blue value
+         */
+        Color(int r, int g, int b);
+
+        /**
+         * @brief Construct a new Color object with default values (black, (0, 0, 0))
```
Comment (AxelF44) [inline include/Color.hpp:24]:
```suggestion
         * @brief Construct a new Color object with default values (black, (0, 0, 0)).
```

File: include/Color.hpp [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+         *
+         */
+        Color() = default;
+
+        /**
+         * @brief Destroy the Color object
```
Comment (AxelF44) [inline include/Color.hpp:30]:
```suggestion
         * @brief Destroy the Color object.
```

**Review by AxelF44 — CHANGES_REQUESTED** [review]

File: include/RaytracerRules.hpp [hunk 4]
```diff
@@ -10,4 +10,6 @@
 
 #pragma once
 
-#define VOID_COLOR Math::Vector3D(0,0,0)
+#include "Color.hpp"
```
Comment (mathieurobert1) [inline include/RaytracerRules.hpp:13]: useless include

File: src/Core/Lights/LightsContainer.cpp [hunk 5]
```diff
@@ -32,5 +32,5 @@ Math::Point3D Light::LightsContainer::computeColor(Math::Vector3D primitiveNorma
             highColor = colorFind;
         }
     }
-    return highColor;
+    return Color(highColor.x(), highColor.y(), highColor.z());
```
Comment (mathieurobert1) [inline src/Core/Lights/LightsContainer.cpp:35]: Use the Color constructor with Vector3D

**Review by mathieurobert1 — CHANGES_REQUESTED** [review]

File: include/Primitives/Cone.hpp [hunk 6]
```diff
@@ -55,7 +55,7 @@ namespace Primitive {
              * @param lights container of lights
              * @return Math::Point3D color
```
Comment (AxelF44) [inline include/Primitives/Cone.hpp:56]:
```suggestion
             * @return Color color class
```

File: include/Primitives/Cylinder.hpp [hunk 7]
```diff
@@ -63,19 +63,19 @@ namespace Primitive {
...
             void setRadius(double radius);
-    
+
             /**
-             * @brief compute the hit point color of a cylinder
+             * @brief Compute the hit point color of a cylinder.
              *
              * @param hitPoint to compute the color
              * @param lights container of lights
              * @return Math::Point3D color
```
Comment (AxelF44) [inline include/Primitives/Cylinder.hpp:72]:
```suggestion
             * @return Color color class
```

**Review by AxelF44 — CHANGES_REQUESTED** [review]

**Review by theophile-jr — DISMISSED** [review]

**Review by AxelF44 — APPROVED** [review]

File: include/Primitives/Cone.hpp
Comment (mathieurobert1) [inline include/Primitives/Cone.hpp:1]: All the brief commentaries of this file should be in Camel case and with a point.

**Review by mathieurobert1 — CHANGES_REQUESTED** [review]

**Review by AxelF44 — APPROVED** [review]

**Review by mathieurobert1 — APPROVED** [review]

**Pull request merged**